docs/source/media.rst

```diff
@@ -0,0 +1,108 @@
+Media API
+---------
+
+.. note:: This article may be moved to the general AIForces doc
+
+Overview
+^^^^^^^^
+The Media API is designed to fetch user uploaded content in a predictable,
+understandable and a secure way. It is also an abstraction layer between
+the user and the internal storage scheme used in the application.
+
+Front-end
+^^^^^^^^^
+All the urls handled by the user have this general shape::
+
+   https://outhosting.com/media/:longidentifier/original_filename.ext
+
+This method has several upsides
+
+1. It stays the same for all file types and uses. From images to submissions
+   and tasks, everything can be retrieved in the same way.
+2. The identifier could be used in many places throughout the application,
+   since it will uniquely map to a particular file.
+3. It allows for pseudo-folders. If all filenames are unique and the
+   circumstances call for that (for example, a task folder) those files can be
+   kept together for ease of access.
+4. It preserves the orignal filenames, which helps the user who uploaded It
+   identify the file later.
+
+Back-end
+^^^^^^^^
+There are basically four parts to the innter workings of this API. The check,
```

Comment:
innter - typo?

Reply:
indeed

Reply:
Will you commit that so that we could resolve the conversation?

Reply:
Fixed in [this commit](https://github.com/AIForces/AIForcesAPI/commit/0257515e3313617b1ecfeb5b929f36024d723577)